### PR TITLE
Refactor for latest Metamask changes

### PR DESCRIPTION
**Pull request by xaler5**
Resolves https://github.com/OpenZeppelin/ethernaut/issues/198 
Resolves https://github.com/OpenZeppelin/ethernaut/issues/200 
and partially https://github.com/OpenZeppelin/ethernaut/issues/188

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -91,17 +89,17 @@ window.addEventListener('load', async() => {
 function checkWrongNetwork(id) {
   let onWrongNetwork = false
   if(constants.ACTIVE_NETWORK.id === constants.NETWORKS.LOCAL.id) {
-    onWrongNetwork = parseInt(id, 10) < 1000
+    onWrongNetwork = Number(id, 10) < 1000
```
Comment (frangio) [inline src/index.js:92]:
```suggestion
    onWrongNetwork = Number(id) < 1000
```

File: src/utils/ethutil.js [hunk 2]
```diff
@@ -7,6 +6,11 @@ export function setWeb3(_web3) {
 }
 
 export function getTruffleContract(jsonABI, defaults = {}) {
+  //Doing this here instead of `import` so that the project uses the project version of Web3Js
```
Comment (wbnns) [inline src/utils/ethutil.js:9]: @xaler5 

Thanks for doing this! Just a couple of nits:

1. Add space after `//`
2. s/Web3Js/web3.js/

Will proceed to test using the preview link.

**Comment by wbnns** [comment]
There are a couple of MetaMask deprecation notices in the console; not sure if it is in scope for this PR, so just mentioning in case helpful:

```
inpage.js:1  MetaMask: The event 'close' is deprecated and may be removed in the future. Please use 'disconnect' instead.
For more information, see: https://eips.ethereum.org/EIPS/eip-1193#disconnect
```

```
inpage.js:1 MetaMask: The event 'data' is deprecated and will be removed in the future. Use 'message' instead.
For more information, see: https://eips.ethereum.org/EIPS/eip-1193#message
```

Related: #188

**Comment by wbnns** [comment]
@xaler5 

Maybe add a note that this "Resolves #198" which will cause GitHub to automatically close it once this is merged.

**Comment by xaler5** [comment]
> There are a couple of MetaMask deprecation notices in the console; not sure if it is in scope for this PR, so just mentioning in case helpful:
> 
> ```
> inpage.js:1  MetaMask: The event 'close' is deprecated and may be removed in the future. Please use 'disconnect' instead.
> For more information, see: https://eips.ethereum.org/EIPS/eip-1193#disconnect
> ```
> 
> ```
> inpage.js:1 MetaMask: The event 'data' is deprecated and will be removed in the future. Use 'message' instead.
> For more information, see: https://eips.ethereum.org/EIPS/eip-1193#message
> ```
> 
> Related: #188

Yes, apparently there's no way right now to get rid of them. They are given by the Web3. Apparently, because of backward compatibility, they decided to not remove them for the time being. You can read more about it [here](https://github.com/MetaMask/metamask-extension/issues/9301) and [here](https://github.com/ChainSafe/web3.js/issues/3699). You can [still see them](https://github.com/ChainSafe/web3.js/pull/3773/files#diff-d25786686c1053b786cc2626dc6e048675050593c0ebaafbf0814e1996f22022R141) being used

**Comment by wbnns** [comment]
@xaler5 

In `Level 0` Step 5:

image

Results in:

image

**Comment by wbnns** [comment]
@xaler5 

I deleted my last comment - apologies for the false notification - I switched off Rinkeby in the middle of testing and forgot to switch back.

File: src/index.js [hunk 3]
```diff
@@ -47,14 +47,12 @@ window.addEventListener('load', async() => {
   if (window.ethereum) {
     window.web3 = new constants.Web3(window.ethereum)
     try {
-      await window.ethereum.enable()
+      await window.ethereum.request({method: `eth_requestAccounts`})
```
Comment (ylv-io) [inline src/index.js:50]: [Metamask docs](https://docs.metamask.io/guide/getting-started.html#connecting-to-metamask):
>You should never initiate a connection request on page load.

Though, it should be address as a separate issue. Let's bring it back ASAP.

**Review by ylv-io — APPROVED** [review]
Hey @xaler5 , thanks for fixing this so fast. Hope it gets pushed to prod fast and we all can enjoy Ethernaut again!

**Comment by frangio** [comment]
Thank you all for working on this and reviewing it!

**Pull request merged**